doc/tdshim_spec.md

```diff
@@ -267,6 +267,14 @@ typedef enum {
   // The entrypoint is defined at HOB_PAYLOAD_INFO_TABLE.Entrypoint.
   PayloadImageTypeRawVmLinux,
 
+  // Payload Binary is vmlinux, follow Linux boot protocol.
+  // It's an ELF64 binary image.
+  PayloadImageTypeVmLinux,
+
+  // Auto-detect type of the payload image, follow Linux boot protocol.
+  // It's one of bzImage, vmLinux. RawVmLinux should be explicitly marked
+  // instead of auto-detection.
+  PayloadImageTypeAutoDetect,
```

Comment:
Should it be align with latest spec in main?

Reply:
Let first merge the spec, then update the code:)

Reply:
i mean the spec should be chanaged - PayloadImageTypeAutoDetect should be removed

Reply:
Sorry, the last patch should be removed:) 
Fixed.